src/plugins/borrow-plugins/plugins/aave/BorrowEngineFactory.js

```diff
@@ -219,22 +230,60 @@ export const makeBorrowEngineFactory = (blueprint: BorrowEngineBlueprint) => {
         })
       },
       async borrow(request: BorrowRequest): Promise<ApprovableAction> {
-        return {
-          networkFee: {
-            currencyCode: wallet.currencyInfo.currencyCode,
-            nativeAmount: '2100000000000000'
-          },
-          approve: async () => {}
-        }
+        const { nativeAmount, tokenId, fromWallet } = asBorrowRequest(request)
+
+        const token = asEdgeToken(tokenId)
+        const tokenAddress = asTokenAddress(token)
+
+        const asset = tokenAddress
+        const amount = BigNumber.from(nativeAmount)
+        const onBehalfOf = (await fromWallet.getReceiveAddress()).publicAddress
+
+        const gasPrice = await aaveNetwork.provider.getGasPrice()
+
+        const borrowTx = await aaveNetwork.lendingPool.populateTransaction.borrow(asset, amount, INTEREST_RATE_MODE, REFERRAL_CODE, onBehalfOf, {
+          gasLimit: '800000',
+          gasPrice
+        })
+
+        return await makeApprovableCall({
+          tx: borrowTx,
+          wallet,
+          token,
+          metadata: {
+            name: 'AAVE',
+            category: 'transfer:borrow',
+            notes: `Borrow ${token.displayName} loan`
+          }
+        })
       },
       async repay(request: RepayRequest): Promise<ApprovableAction> {
-        return {
-          networkFee: {
-            currencyCode: wallet.currencyInfo.currencyCode,
-            nativeAmount: '2100000000000000'
-          },
-          approve: async () => {}
-        }
+        const { nativeAmount, tokenId, fromWallet } = asRepayRequest(request)
+
+        const token = asEdgeToken(tokenId)
+        const tokenAddress = asTokenAddress(token)
+
+        const asset = tokenAddress
+        const amount = BigNumber.from(nativeAmount)
+        const onBehalfOf = (await fromWallet.getReceiveAddress()).publicAddress
+
+        const gasPrice = await aaveNetwork.provider.getGasPrice()
+
+        const repayTx = await aaveNetwork.lendingPool.populateTransaction.repay(asset, amount, INTEREST_RATE_MODE, onBehalfOf, {
+          gasLimit: '800000',
+          gasPrice
+        })
+
+        return await makeApprovableCall({
```

Comment:
We don't need an approval for this? How do we return the USDC we borrowed earlier, without sending it back to AAVE?

Reply:
It didn't mention so in the documentation for this function (https://docs.aave.com/developers/v/2.0/the-core-protocol/lendingpool#repay) though it would make sense that this is required... Perhaps it omitted it because it is assumed that the borrower account has already approved by the `deposit` method. That assumption isn't guaranteed, especially if this is invoked not by the borrower but by an account repaying on behalf of the borrower.

I can add the approve just to be sure.